src/main/java/org/sefglobal/scholarx/service/ProgramService.java

```diff
@@ -154,4 +161,44 @@ public void deleteProgram(long id) throws ResourceNotFoundException {
         }
         return mentorRepository.findAllByProgramId(id);
     }
+
+    /**
+     * Create new {@link Mentor}
+     *
+     * @param programId which is the program id for the requesting {@link Program}
+     * @param profileId which is the profile id of the applying user's {@link Profile}
+     * @param mentor    which holds the data to be added
+     * @return the created {@link Mentor}
+     * @throws ResourceNotFoundException is thrown if the applying {@link Program} doesn't exist
+     * @throws ResourceNotFoundException is thrown if the applying user's {@link Profile} doesn't exist
+     * @throws BadRequestException is thrown if the applying {@link Program} is not in applicable {@link ProgramState}
+     */
+    public Mentor applyAsMentor(long programId, long profileId, Mentor mentor) throws ResourceNotFoundException, BadRequestException {
+        Optional<Program> optionalProgram = programRepository.findById(programId);
+        if (!optionalProgram.isPresent()) {
+            String msg = "Error, Unable to apply as a mentor. " +
+                         "Program with id: " + programId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+        if (!optionalProgram.get().getState().equals(ProgramState.MENTOR_APPLICATION)) {
```

Comment:
If getState() return null a Null pointer exception will be thrown. 
https://stackoverflow.com/a/2358029
```suggestion
        if (!ProgramState.MENTOR_APPLICATION.equals(optionalProgram.get().getState())) {
```

Reply:
Ah right. I'll fix it

Reply:
Fixed